src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http3RequestStream.cs

```diff
@@ -139,6 +142,11 @@ public async Task<HttpResponseMessage> SendAsync(CancellationToken cancellationT
             bool shouldCancelBody = true;
             try
             {
+                Http3WebtransportSession? webtransportSession = null;
+                if (_request.IsWebTransportH3Request)
```

Comment:
nit: you are calling the property `IsWebTransportH3Request` several times; and the property will perform all the operations incl. string comparison each time. You may want to evaluate it once and cache the value, e.g. in `private bool _isWebtransportSessionStream` that you had.